charts/platform-system/templates/flux/istio/_istio.tpl

```diff
@@ -0,0 +1,65 @@
+{{ define "platform-system.istio.repository.name" }}
+{{- printf "%s-%s" .Release.Name .Values.apps.istio.chart.source.repository | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end -}}
+
+{{/*
+  Istio base templates
+*/}}
+
+{{ define "platform-system.istio.istio-base.release.name" }}
+{{- printf "%s-istio-base" .Release.Name | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end -}}
+
+{{- define "platform-system.istio.istio-base.defaultValues" -}}
+{{- $istioValues := .Values.apps.istio }}
+global:
+  istioNamespace: {{ $istioValues.namespace }}
+{{- end -}}
+
+{{- define "platform-system.istio.istio-base.values" }}
+{{- $defaultValues := (include "platform-system.istio.istio-base.defaultValues" $ | fromYaml ) }}
+{{- (deepCopy .Values.apps.istio.component.base.values | merge $defaultValues) | toYaml }}
+{{- end -}}
+
+{{/*
+  Istio discovery templates
+*/}}
+
+{{- define "platform-system.istio.istiod.defaultValues" -}}
+{{- $istioValues := .Values.apps.istio }}
+global:
+  istioNamespace: {{ $istioValues.namespace }}
+pilot:
+  podAnnotations: {{ include "helper.appAnnotations" (dict "appAnnotations" .Values.apps.istio.annotations) | indent 4 }}
+  nodeSelector: {{ include "helper.nodeSelector" (dict "appNodeSelector" .Values.apps.istio.nodeSelector "globalNodeSelector" .Values.global.nodeSelector ) | indent 4 }}
+  tolerations: {{ include "helper.tolerations" (dict "appTolerations" .Values.apps.istio.tolerations "globalTolerations" .Values.global.tolerations ) | indent 4 }}
+{{- end -}}
+
+{{ define "platform-system.istio.istiod.release.name" }}
```

Comment:
Same as above. This name thing should just take a list of things to join on `-`. This is too repetitive for what's basically just a join.